### PR TITLE
[FIX] Fixing all the failing test cases and adding missing test cases

### PR DESCRIPTION
**Pull request by ghost**
1. Remove all the unused files from the code base including (lib/ and
   spec/)
2. Fixed existing test cases to work with the lastest code at
   api.ripple.com
3. Added new test cases "get_transaction", "get_transaction_fee",
   "get_server_status", "get_balance" and "generate_uuid".
4. Added fixtures to make sure that all the expected fields from the
   response exist.
5. Cleaned up package.json file to remove all the unused dependencies.

### CODE REVIEW

**Comment by nerdylocks** [comment]
Review in progress

File: test/get_transaction.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      payments = payment_array;
+      done();
+    });
+  });
+
+  // Given the hash for the second last payment, the response should contain the
+  // hashes of last payment and third last payment.
+  it('should get notifications', function(done) {
+    this.timeout(5000);
+    client.getTransaction(payments[0].hash, function(error, response) {
```
Comment (cdm6a) [inline test/get_transaction.js:28]: Might want to double check this but if more payments get added to that particular wallet it's possible that payments[0] might be different in the future.

File: test/send_and_confirm_payment.js [hunk 2]
```diff
@@ -84,15 +85,14 @@ describe('Ripple REST Client sendPayment', function() {
         secret: SECRET
       };
 
-      client.sendAndConfirmPayment(paymentObj, function(error, response){
-        assert(error);
-        assert.strictEqual(error.error_type, 'invalid_request');
+      client.sendAndConfirmPayment(paymentObj, function(error, response) {
+        assert(!response);
+        assert(error.response.body);
+        assert.strictEqual(error.response.body.error_type, 'invalid_request');
```
Comment (cdm6a) [inline test/send_and_confirm_payment.js:91]: Probably should check the additional error field on the body too?

File: test/set_trust_lines.js [hunk 3]
```diff
@@ -31,18 +35,21 @@ describe('Ripple REST Client setTrustLines', function() {
...
-      issuer: fixtures.ripple_address.source_account
+      issuer: account_info.source_account
     };
 
-    client.setTrustLines(trust, function(error, response){
+    client.setTrustLines(trust, function(error, response) {
+      assert(error, 'Error is null');
+      assert(error.response.body.success === false, 'Request should fail');
+      assert.strictEqual(error.response.body.error_type, 'invalid_request');
+      assert.strictEqual(error.response.body.error, 'restINVALID_PARAMETER');
```
Comment (cdm6a) [inline test/set_trust_lines.js:50]: ^Like here.

File: test/get_transaction.js (hunk 1 above)
Comment (nerdylocks) [inline test/get_transaction.js:28]: This looks fine. It gets the latest payment from the notification url.

File: test/fixtures/get_balance.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+exports.rootExpectedKeys = [
+	'success', 'ledger', 'validated', 'balances'
```
Comment (nerdylocks) [inline test/fixtures/get_balance.js:4]: I see your intent here with these kind of fixtures. However, if the structure of response changes (i.e new property with child properties) we won't be able to know. My suggestions is to mimic the json object response and asserting with a deepEqual, like you have in  `test/fixtures/get_notification.js`

File: test/get_transaction.js (hunk 1 above)
Comment (ghost) [inline test/get_transaction.js:28]: Abiy is correct. We are always getting the latest payment.

File: test/fixtures/get_balance.js (hunk 4 above)
Comment (ghost) [inline test/fixtures/get_balance.js:4]: Will unify there structures of the fixtures on the next pull request

File: test/send_and_confirm_payment.js (hunk 2 above)
Comment (ghost) [inline test/send_and_confirm_payment.js:91]: Will add in the next pull request.

File: index.js [hunk 5]
```diff
@@ -304,7 +317,12 @@ Client.prototype._getAndHandlePaymentStatus = function(statusUrl, callback, loop
 Client.prototype.pollPaymentStatus = function(payment, callback){
   var self = this;
   if (payment && payment.status_url) {
-    self._getAndHandlePaymentStatus(payment.status_url, callback, self._getAndHandlePaymentStatus.bind(this));
+    var url = payment.status_url;
+
+    if (url.indexOf('http') < 0) {
```
Comment (nerdylocks) [inline index.js]: you could also accomplish this like below. A bit cleaner, imo.

```
var urlParse = require('url').parse;
...
var url = urlParse(payment.status_url) ;
if (_.isEmpty(url.protocol)) {
...
```

File: test/update_account_settings.js [hunk 6]
```diff
@@ -33,4 +32,22 @@ describe('updating account settings', function() {
...
+    };
+
+    client.updateAccountSettings(optsHotWallet, function(error, settings) {
+      assert(error);
+      assert(!settings);
+      assert(!error.response.body.success);
+      assert.strictEqual(error.response.body.error_type, 'invalid_request');
+      assert.strictEqual(error.response.body.error, 'restINVALID_PARAMETER');
+      assert.strictEqual(error.response.body.message, 'Parameter missing: secret');
+      done();
```
Comment (nerdylocks) [inline test/update_account_settings.js:49]: @ZhenyuRipple , thanks for adding error cases.

File: index.js (hunk 5 above)
Comment (ghost) [inline index.js]: Thanks for the great info!  Will change in the next iteration.

**Pull request closed without merging**